### PR TITLE
Move Makefile and docker-compose.yml to repo root

### LINKED ISSUE #1511: chore: move Makefile to repo root, out of docker directory
Our Makefile should be in the repo root as this is the expected location for a project Makefile.

In order to facilitate this change, various paths in the following files will probably need to be changed to make this work

- [x] Makefile itself
- [x] GHA yml files
- [x] Dockerfile in various folders?
- [x] docker-compose.yml
- [x] our various README files

As a matter of opinion, it would also seem appropriate to move the docker-compose.yml file to the repo root as well.

Thoughts @longrunningprocess ?

### PR DESCRIPTION
**Pull request by myieye**
Fixes #1511

## Description

Makefiles as well as docker-compose.yml are expected to be in the project root, so this PR moves them both there.

### Type of Change

- Repo restructure

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message

### CODE REVIEW

**Review by megahirt — APPROVED** [review]
wow, this is great work with surgical precision!  LGTM

I assume that you have run all of the Makefile targets locally and it behaves as expected, since CI does not exercise all targets.

**Comment by myieye** [comment]
> wow, this is great work with surgical precision! LGTM
> 
> I assume that you have run all of the Makefile targets locally and it behaves as expected, since CI does not exercise all targets.

Yeah, I ran all the make targets in the main makefile. `e2e-tests` fails, but I'm pretty sure that's unrelated/expected.
```
Bellows E2E Change Password app
    ✓ setup: login as user, go to change password page
    ✓ refuses to allow form submission if the confirm input does not match
    ✓ allows form submission if the confirm input matches
    ✓ can successfully changes user's password after form submission

  Bellows E2E Project Settings app
    ✗ Normal user cannot access projectSettings to a project of which the user is a member
      - Failed: Project Test Project not found.
          at wrappedReject (/data/node_modules/jasminewd2/index.js:100:24)
          at runMicrotasks (<anonymous>)
          at processTicksAndRejections (node:internal/process/task_queues:96:5)
      From: Task: Run it("Normal user cannot access projectSettings to a project of which the user is a member") in control flow
          at UserContext.<anonymous> (/data/node_modules/jasminewd2/index.js:94:19)
      From asynchronous test: 
      Error
          at Suite.<anonymous> (/data/test/app/bellows/project-settings.e2e-spec.ts:14:3)
          at apply (/data/node_modules/lodash/lodash.js:490:27)
          at Env.wrapper [as describe] (/data/node_modules/lodash/lodash.js:5379:16)
          at Object.<anonymous> (/data/test/app/bellows/project-settings.e2e-spec.ts:8:1)
          at Module._compile (node:internal/modules/cjs/loader:1103:14)
```

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "webpack:prd": "webpack --config webpack-prd.config.js",
     "compile-test-e2e": "tsc -p test/app",
     "test-e2e": "protractor test/app/protractorConf.js",
-    "make": "cd docker && make",
+    "make": "make",
```
Comment (longrunningprocess) [inline package.json:19]: I don't think we need this anymore

File: .vscode/tasks.json [hunk 2]
```diff
@@ -53,11 +53,8 @@
     {
       "label": "Reset E2E tests",
       "type": "process",
-      "command": "docker-compose",
-      "args": ["restart", "app-for-e2e"],
-      "options": {
-        "cwd": "docker"
-      },
+      "command": "docker",
```
Comment (longrunningprocess) [inline .vscode/tasks.json:56]: not sure if this is viable but I'd like to see all "commands" run through make targets if possible, do you think that's possible here too?

File: docker-compose.yml [hunk 3]
```diff
@@ -13,20 +13,20 @@ services:
...
-      - ../src/sass:/data/src/sass
-      - ../src/Site/views/languageforge/theme/default/sass:/data/src/Site/views/languageforge/theme/default/sass
+      - ./src/angular-app:/data/src/angular-app
+      - ./src/sass:/data/src/sass
+      - ./src/Site/views/languageforge/theme/default/sass:/data/src/Site/views/languageforge/theme/default/sass
 
   app:
     build:
-      context: ..
+      context: .
```
Comment (longrunningprocess) [inline docker-compose.yml:29]: did you check whether this is actually required now?  I might expect that to be the default.

File: docker-compose.yml [hunk 4]
```diff
@@ -175,7 +175,7 @@ services:
 
   next-app:
     build:
-      context: ../next-app
+      context: next-app
       dockerfile: ../docker/next-app/Dockerfile.next-app
```
Comment (longrunningprocess) [inline docker-compose.yml:179]: this seems like it should be `./docker...`, no?

File: docs/DEVELOPER.md [hunk 5]
```diff
@@ -46,13 +46,13 @@ On Windows, the project should be opened with the [Remote - WSL](https://marketp
...
 
-> Note: `make` rules can be run directly from the `docker` directory or with `npm run make` from any project sub-directory.
+> Note: `make` rules can be run directly from the root directory or with `npm run make` from any project sub-directory.
 
-1. `npm run make` or `cd docker && make`
-1. Within any browser, navigate to https://localhost
-1. Continue through any certificate warnings
-1. You should see a landing page, click "Login"
-1. Use `admin` and `password` to login
+1. `make` or `npm run make`
```
Comment (longrunningprocess) [inline docs/DEVELOPER.md:51]: I'd prefer to just one one way to do this and remove the `npm run make`

File: playwright.config.ts [hunk 6]
```diff
@@ -101,7 +101,7 @@ const config: PlaywrightTestConfig = {
 
   /* Run your local dev server before starting the tests */
   webServer: {
-    command: 'docker-compose -f docker/docker-compose.yml up app-for-playwright',
+    command: 'docker-compose -f docker-compose.yml up app-for-playwright',
```
Comment (longrunningprocess) [inline playwright.config.ts:104]: you probably don't need this `-f ...` at all anymore.

**Review by longrunningprocess — COMMENTED** [review]
excellent work so far, I think there are a handful of improvements we can still make

File: docker-compose.yml (hunk 4 above)
Comment (myieye) [reply docker-compose.yml:179]: It's relative to the context, so it needs `..` the way it's currently set up.

File: docker-compose.yml (hunk 3 above)
Comment (myieye) [reply docker-compose.yml:29]: Using simply: `build: path/to/Dockerfile-folder` seems to set the context to that folder, which causes lots of trouble in the `Dockerfile`s, so I'm just going to remove the meaningless `context` properties as suggested here.

**Pull request merged**